cmd/juju/application/deploy.go

```diff
@@ -1650,12 +1653,38 @@ func (c *DeployCommand) charmStoreCharm() (deployFn, error) {
 	}, nil
 }
 
+// Returns the first string that isn't empty.
+// If all strings are empty, then return an empty string.
+func getPotentialSeriesName(series ...string) string {
+	for _, s := range series {
+		if s != "" {
+			return s
+		}
+	}
+	return ""
+}
+
 // validateCharmSeriesWithName calls the validateCharmSeries, but handles the
 // error return value to check for NotSupported error and returns a custom error
 // message if that's found.
 func (c *DeployCommand) validateCharmSeriesWithName(series, name string) error {
-	if err := c.validateCharmSeries(series); err != nil {
+	err := c.validateCharmSeries(series)
+	return charmValidationError(series, name, errors.Trace(err))
+}
+
+// charmValidationError consumes an error along with a charmSeries and name
+// to help provide better feedback to the user when somethings gone wrong around
+// validating a charm validation
+func charmValidationError(charmSeries, name string, err error) error {
+	if err != nil {
 		if errors.IsNotSupported(err) {
+			if warningInfo := series.SeriesWarningInfo(charmSeries); len(warningInfo) > 0 {
+				logger.Warningf(
+					"Parsing issues occured when extracting the distro-info.\n"+
```

Comment:
s/occured/occurred/  or is that a british-ism?

Reply:
dunno, internet isn't telling me :D